addon/record-data.js

```diff
@@ -692,7 +692,14 @@ export default class FragmentRecordData extends RecordData {
       if (this._fragments[key]) {
         continue;
       }
-      if ((updates[key] === null) !== (original[key] === null)) {
+      const wasOrBecomingNull = original[key] === null || updates[key] === null;
+      let isChangedArray = original[key]?.length || updates[key]?.length;
+      if (isChangedArray) {
+        isChangedArray = !updates[key]?.filter?.(
+          (record) => !original[key]?.includes(record.clientId)
+        );
```

Comment:
Could the updates have an entry for a new value? In that case this wouldn't be marked as changed, right?

Reply:
This first attempt has some issues. Looking at this a bit more, I see we use `diffArray` in `addons/array/stateful.js`. Maybe we should use that here as well?

Reply:
Unfamiliar with it, but all for reuse.

Reply:
Used that and added a test as well. Tested it with the example in #467 and it worked there as well.